#!/bin/sh
patch -p1 -f $* < $0
exit $?

From ca07f7ae44ac9476fa07a538b098a968fed92bdc Mon Sep 17 00:00:00 2001
From: Jonathan Boeing <jonathan.n.boeing@gmail.com>
Date: Wed, 25 Nov 2020 20:19:46 -0700
Subject: [PATCH 1/2] Use fclose() instead of close()

FILE *file is opened with fdopen(), so close with fclose()

Fixes these compiler warnings:
warning: passing argument 1 of 'close' makes integer from pointer without a cast [-Wint-conversion]
---
 src/quote_fmt_parse.y | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/quote_fmt_parse.y b/src/quote_fmt_parse.y
index a7bed9081..d8226feae 100644
--- a/src/quote_fmt_parse.y
+++ b/src/quote_fmt_parse.y
@@ -528,7 +528,7 @@ static void quote_fmt_insert_program_output(const gchar *progname)
 		while (fgets(buffer, sizeof(buffer), file)) {
 			INSERT(buffer);
 		}
-		close(file);
+		fclose(file);
 	}
 }
 
@@ -575,7 +575,7 @@ static void quote_fmt_attach_file_program_output(const gchar *progname)
 			strretchomp(buffer);
 			attachments = g_list_append(attachments, g_strdup(buffer));
 		}
-		close(file);
+		fclose(file);
 	}
 }
 
-- 
2.20.1

